plugin.video.fsgo/resources/lib/fsgo.py

```diff
@@ -0,0 +1,379 @@
+﻿# -*- coding: utf-8 -*-
+"""
+A Kodi-agnostic library for FOX Sports GO
+"""
+import json
+import codecs
+import cookielib
+import time
+from datetime import datetime, timedelta
+import calendar
+import uuid
+from urllib import urlencode
+
+import requests
+import m3u8
+import iso8601
+
+
+class fsgolib(object):
+    def __init__(self, cookie_file, credentials_file, debug=False, verify_ssl=True):
+        self.debug = debug
+        self.verify_ssl = verify_ssl
+        self.http_session = requests.Session()
+        self.cookie_jar = cookielib.LWPCookieJar(cookie_file)
+        self.credentials_file = credentials_file
+        self.base_url = 'https://media-api.foxsportsgo.com'
+        try:
+            self.cookie_jar.load(ignore_discard=True, ignore_expires=True)
+        except IOError:
+            pass
+        self.http_session.cookies = self.cookie_jar
+
+    class LoginFailure(Exception):
+        def __init__(self, value):
+            self.value = value
+
+        def __str__(self):
+            return repr(self.value)
+
+    def log(self, string):
+        if self.debug:
+            try:
+                print '[fsgolib]: %s' % string
```

Comment:
@enen92: This lib is supposed to be able to run independent of Kodi, which means that I can't import any xbmc-related modules in it. This has not been a problem for a few other add-ons I have submitted.